src/python/WMCore/WMRuntime/Tools/Scram.py

```diff
@@ -110,7 +110,9 @@ def preCommand(self):
 
         """
         result = ""
-        if self.architecture != None:
+        if self.architecture == None:
```

Comment:
Tried to make it pick the architecture automatically. This works interactively, but not here unfortunately. Have HN thread open about it.